ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -1556,11 +1548,29 @@ enum pkcs11_rc check_attrs_against_modification(struct pkcs11_session *session,
 			}
 		}
 
-		if (cli_ref.id == PKCS11_CKA_PRIVATE &&
-		    get_bool(head, PKCS11_CKA_PRIVATE)) {
-			if ((pkcs11_session_is_public(session) ||
-			     pkcs11_session_is_so(session)))
-				return PKCS11_CKR_USER_NOT_LOGGED_IN;
+		if (cli_ref.id == PKCS11_CKA_PRIVATE) {
+			bool parent_priv =
+				get_bool(obj->attributes, cli_ref.id);
+			uint8_t bbool = 0;
+			uint32_t size = sizeof(bbool);
+			get_attribute(head, cli_ref.id, &bbool, &size);
```

Comment:
could use `get_bool(head, cli_ref.id)`.

Reply:
Sure, I will change that. I didn't use it as it also returns FALSE when attribute is not found. But now I realize that in this case, we are calling it only if attribute is present in head, so attribute not found case will never be there.